### ISSUE TITLE
Input issue with Facebook when input area is empty

### ISSUE THREAD

**Issue by ekangmonyet** [chewing/ibus-chewing#227]
### Problem
Committed text is not "sent" to the facebook messenger chat input area if it is empty

This is highly specific to a particular situation. If the input area already has text in it, everything works fine. However, if the input area is empty, and is in Chinese input mode (pre-edit and lookup table works fine), once the pre-edit buffer is committed (by pressing and releasing ENTER key), the text is gone. It is probably not a visual glitch as pressing ENTER multiple time wont send the previously committed text even if it's not visible.

**What works:**
* Input area of messenger chat, facebook post and facebook comment, when input area is not empty
* Same typing behavior in Facebook search area (both global search and messenger chat search works),
* Same typing behavior in almost all other sites (such as Github, duckduckgo), no other site with same issue found yet,
* Other input method with pre-edit (tested: anthy, bopomofo)

**What doesn't:**
* Input area of messenger chat, facebook post and facebook comment, when input area is empty

### Environment
Firefox 129.0 (also tested with chromium 127.0)
Gnome 46 with X11
ibus 1.5.30

### Debug Log
Log between ENTER press to release, until the `outgoingStr` is empty. The text `可機` is used for testing.

When input area is empty (input with issue):
```
ibus-chewing-INFO: 01:26:34.331: ***** ibus_chewing_pre_edit_process_key(-,ff0d(Return),10(MOD2))
(ibus-engine-chewing:96015): MKDG-DEBUG: 01:26:34.331: property_context_get(plain-zhuyin): value=0
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.331: ibus_chewing_pre_edit_process_key(): Before response flags=0 buff_check=1 bpmf_check=0 cursor=2 total_choice=0 is_chinese=1 is_full_shape=0 is_plain_zhuyin=0
ibus-chewing-INFO: 01:26:34.331: * self_handle_return(-,ff0d(Return),10(MOD2))
ibus-chewing-INFO: 01:26:34.331: commitStr=|可機|

ibus-chewing-INFO: 01:26:34.331: outgoing=|可機|

(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.331: ibus_chewing_pre_edit_update_outgoing(-): return: outgoing=|可機|
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.331: ibus_chewing_pre_edit_process_key() response=0
(ibus-engine-chewing:96015): MKDG-DEBUG: 01:26:34.331: property_context_get(plain-zhuyin): value=0
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.331: ibus_chewing_pre_edit_process_key(): After response flags=8 buff_check=0 bpmf_check=0 cursor=0 total_choice=0 is_chinese=1 is_full_shape=0 is_plain_zhuyin=0
(ibus-engine-chewing:96015): MKDG-DEBUG: 01:26:34.331: property_context_get(plain-zhuyin): value=0
(ibus-engine-chewing:96015): MKDG-DEBUG: 01:26:34.331: property_context_get(plain-zhuyin): value=0
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.331: ibus_chewing_pre_edit_process_key(): After plain-zhuyin handling flags=8 buff_check=0 bpmf_check=0 cursor=0 total_choice=0 is_chinese=1 is_full_shape=0 is_plain_zhuyin=0
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.331: * ibus_chewing_pre_edit_update(-)
ibus-chewing-INFO: 01:26:34.331: * ibus_chewing_pre_edit_update(-)  bufferStr=||, bpmfStr=|| bpmfLen=0 cursor=0
(ibus-engine-chewing:96015): MKDG-DEBUG: 01:26:34.331: property_context_get(sel-keys): value=1234567890
(ibus-engine-chewing:96015): MKDG-DEBUG: 01:26:34.331: property_context_get(cand-per-page): value=10
ibus-chewing-INFO: 01:26:34.331: ***** ibus_chewing_lookup_table_update(): choicePerPage=0, totalChoice=0, currentPage=0
ibus-chewing-INFO: 01:26:34.331: ibus_chewing_pre_edit_process_key() candidateCount=0
ibus-chewing-Message: 01:26:34.331: process_key_event() result=1
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.331: update() statusFlags=17
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.331: decorate_pre_edit() cursor=0 preEdit= charLen=0
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.331: update_aux_text()
ibus-chewing-INFO: 01:26:34.331: refresh_aux_text()
ibus-chewing-INFO: 01:26:34.331: refresh_outgoing_text() outgoingStr=|可機|
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.331: refresh_outgoing_text() outgoingText=|可機|
ibus-chewing-Message: 01:26:34.331: * parent_commit_text(-): outgoingText=可機
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.331: ibus_chewing_pre_edit_clear_outgoing(-)
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.331: update() nPhoneSeq=0 statusFlags=17
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.331: update_lookup_table() CurrentPage=0
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.331: refresh_property(InputMode) status=17
(ibus-engine-chewing:96015): MKDG-DEBUG: 01:26:34.331: property_context_get(show-systray): value=1
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.331: refresh_property(chewing_alnumSize_prop) status=17
(ibus-engine-chewing:96015): MKDG-DEBUG: 01:26:34.331: property_context_get(show-systray): value=1
ibus-chewing-INFO: 01:26:34.414: ******** process_key_event(-,ff0d(Return),1c,40000010) MOD2| RELEASE
(ibus-engine-chewing:96015): MKDG-DEBUG: 01:26:34.414: mkdg_g_settings_read_value(-,-,use-system-keyboard-layout)
ibus-chewing-INFO: 01:26:34.414: ***** ibus_chewing_pre_edit_process_key(-,ff0d(Return),40000010(MOD2| RELEASE))
(ibus-engine-chewing:96015): MKDG-DEBUG: 01:26:34.414: property_context_get(plain-zhuyin): value=0
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.414: ibus_chewing_pre_edit_process_key(): Before response flags=8 buff_check=0 bpmf_check=0 cursor=0 total_choice=0 is_chinese=1 is_full_shape=0 is_plain_zhuyin=0
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.414: ibus_chewing_pre_edit_process_key() response=2
(ibus-engine-chewing:96015): MKDG-DEBUG: 01:26:34.414: property_context_get(plain-zhuyin): value=0
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.414: ibus_chewing_pre_edit_process_key(): After response flags=8 buff_check=0 bpmf_check=0 cursor=0 total_choice=0 is_chinese=1 is_full_shape=0 is_plain_zhuyin=0
ibus-chewing-Message: 01:26:34.414: process_key_event() result=0
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.415: update() statusFlags=17
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.415: decorate_pre_edit() cursor=0 preEdit= charLen=0
(ibus-engine-chewing:96015): ibus-chewing-DEBUG: 01:26:34.415: update_aux_text()
ibus-chewing-INFO: 01:26:34.415: refresh_aux_text()
ibus-chewing-INFO: 01:26:34.415: refresh_outgoing_text() outgoingStr=||
```

When input area is not empty (input without issue):
```
ibus-chewing-INFO: 01:24:06.623: ***** ibus_chewing_pre_edit_process_key(-,ff0d(Return),10(MOD2))
(ibus-engine-chewing:95796): MKDG-DEBUG: 01:24:06.623: property_context_get(plain-zhuyin): value=0
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.623: ibus_chewing_pre_edit_process_key(): Before response flags=0 buff_check=1 bpmf_check=0 cursor=2 total_choice=0 is_chinese=1 is_full_shape=0 is_plain_zhuyin=0
ibus-chewing-INFO: 01:24:06.623: * self_handle_return(-,ff0d(Return),10(MOD2))
ibus-chewing-INFO: 01:24:06.624: commitStr=|可機|

ibus-chewing-INFO: 01:24:06.624: outgoing=|可機|

(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.624: ibus_chewing_pre_edit_update_outgoing(-): return: outgoing=|可機|
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.624: ibus_chewing_pre_edit_process_key() response=0
(ibus-engine-chewing:95796): MKDG-DEBUG: 01:24:06.624: property_context_get(plain-zhuyin): value=0
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.624: ibus_chewing_pre_edit_process_key(): After response flags=8 buff_check=0 bpmf_check=0 cursor=0 total_choice=0 is_chinese=1 is_full_shape=0 is_plain_zhuyin=0
(ibus-engine-chewing:95796): MKDG-DEBUG: 01:24:06.624: property_context_get(plain-zhuyin): value=0
(ibus-engine-chewing:95796): MKDG-DEBUG: 01:24:06.624: property_context_get(plain-zhuyin): value=0
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.624: ibus_chewing_pre_edit_process_key(): After plain-zhuyin handling flags=8 buff_check=0 bpmf_check=0 cursor=0 total_choice=0 is_chinese=1 is_full_shape=0 is_plain_zhuyin=0
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.624: * ibus_chewing_pre_edit_update(-)
ibus-chewing-INFO: 01:24:06.624: * ibus_chewing_pre_edit_update(-)  bufferStr=||, bpmfStr=|| bpmfLen=0 cursor=0
(ibus-engine-chewing:95796): MKDG-DEBUG: 01:24:06.624: property_context_get(sel-keys): value=1234567890
(ibus-engine-chewing:95796): MKDG-DEBUG: 01:24:06.624: property_context_get(cand-per-page): value=10
ibus-chewing-INFO: 01:24:06.624: ***** ibus_chewing_lookup_table_update(): choicePerPage=0, totalChoice=0, currentPage=0
ibus-chewing-INFO: 01:24:06.624: ibus_chewing_pre_edit_process_key() candidateCount=0
ibus-chewing-Message: 01:24:06.624: process_key_event() result=1
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.624: update() statusFlags=17
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.624: decorate_pre_edit() cursor=0 preEdit= charLen=0
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.624: update_aux_text()
ibus-chewing-INFO: 01:24:06.624: refresh_aux_text()
ibus-chewing-INFO: 01:24:06.624: refresh_outgoing_text() outgoingStr=|可機|
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.624: refresh_outgoing_text() outgoingText=|可機|
ibus-chewing-Message: 01:24:06.624: * parent_commit_text(-): outgoingText=可機
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.624: ibus_chewing_pre_edit_clear_outgoing(-)
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.624: update() nPhoneSeq=0 statusFlags=17
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.624: update_lookup_table() CurrentPage=0
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.624: refresh_property(InputMode) status=17
(ibus-engine-chewing:95796): MKDG-DEBUG: 01:24:06.624: property_context_get(show-systray): value=1
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.624: refresh_property(chewing_alnumSize_prop) status=17
(ibus-engine-chewing:95796): MKDG-DEBUG: 01:24:06.624: property_context_get(show-systray): value=1
ibus-chewing-INFO: 01:24:06.708: ******** process_key_event(-,ff0d(Return),1c,40000010) MOD2| RELEASE
(ibus-engine-chewing:95796): MKDG-DEBUG: 01:24:06.708: mkdg_g_settings_read_value(-,-,use-system-keyboard-layout)
ibus-chewing-INFO: 01:24:06.708: ***** ibus_chewing_pre_edit_process_key(-,ff0d(Return),40000010(MOD2| RELEASE))
(ibus-engine-chewing:95796): MKDG-DEBUG: 01:24:06.708: property_context_get(plain-zhuyin): value=0
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.708: ibus_chewing_pre_edit_process_key(): Before response flags=8 buff_check=0 bpmf_check=0 cursor=0 total_choice=0 is_chinese=1 is_full_shape=0 is_plain_zhuyin=0
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.708: ibus_chewing_pre_edit_process_key() response=2
(ibus-engine-chewing:95796): MKDG-DEBUG: 01:24:06.708: property_context_get(plain-zhuyin): value=0
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.708: ibus_chewing_pre_edit_process_key(): After response flags=8 buff_check=0 bpmf_check=0 cursor=0 total_choice=0 is_chinese=1 is_full_shape=0 is_plain_zhuyin=0
ibus-chewing-Message: 01:24:06.708: process_key_event() result=0
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.708: update() statusFlags=17
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.708: decorate_pre_edit() cursor=0 preEdit= charLen=0
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.708: update_aux_text()
ibus-chewing-INFO: 01:24:06.708: refresh_aux_text()
ibus-chewing-INFO: 01:24:06.708: refresh_outgoing_text() outgoingStr=||
(ibus-engine-chewing:95796): ibus-chewing-DEBUG: 01:24:06.708: refresh_outgoing_text() outgoingText=||
```

Unless I missed something, the logged output seems to be identical for both situation. Currently I am clueless, happy to chime in anything I can do/provide to help fixing this issue.

**Comment by kanru** [comment]
Thanks for the report. Could you also provide the version of ibus-chewing and libchewing?

For ibus-chewing >= 2.1.0 please also provide the troubleshooting information from IBus Chewing Preferences > About

```
/usr/libexec/ibus-setup-chewing --about
```

For ibus-chewing < 2.1.0 please include the output of this command

```
gsettings list-recursively org.freedesktop.IBus.Chewing
```

**Comment by kanru** [comment]
I can't reproduce with Debian testing
- GNOME Classic on Xorg
- Firefox 129
- ibus 1.5.30
- ibus-chewing 2.0.0

**Comment by ekangmonyet** [comment]
I didn't realize Fedora 40 is still on ibus-chewing 1.6, I've updated to 2.10 via Rawhide but unfortunately it still does not work. Updated environment tested:
* Firefox 129.0 (also tested with chromium 127.0)
* Gnome 46 with X11 (classic)
* ibus 1.5.30
* ibus-chewing 2.1.0 / 1.6.1  
* libchewing 0.9.0 / 0.6.0

Troubleshooting info:
```
ibus-chewing: 2.1.0-

Compiled against:
- ibus: 1.5.31-beta1
- libchewing: 0.9.0

Running against:
- libchewing: 0.9.0

Settings:
- kb-type: default
- sel-keys: 1234567890
- plain-zhuyin: 0
- conversion-engine: 1
- auto-shift-cur: 1
- add-phrase-direction: 1
- clean-buffer-focus-out: 0
- easy-symbol-input: 1
- esc-clean-all-buf: 0
- enable-fullwidth-toggle-key: 1
- max-chi-symbol-len: 20
- chi-eng-mode-toggle: shift
- sync-caps-lock: disable
- default-english-case: lowercase
- cand-per-page: 10
- show-page-number: 0
- phrase-choice-from-last: 1
- space-as-selection: 0
- vertical-lookup-table: 0
- notify-mode-change: 1

Environment:
- Desktop: GNOME-Classic:GNOME
- Session: gnome-classic-xorg (x11)
- Language: en_US.UTF-8
```

**Comment by kanru** [comment]
I just reproduced in Fedora 40, interesting..

**Comment by kanru** [comment]
It's only reproducible when running in X11. I suspect this is an ibus x11 module issue.

**Comment by ekangmonyet** [comment]
Thanks for your time! I suspected that too, but I wonder why the others like anthy and bopomofo works fine.

**Comment by kanru** [comment]
Yeah, I'll dig more into this.

**Comment by kanru** [comment]
It seems to be a ~~regression~~ change related to #201
For some reason the Facebook code expect the preedit text to exist while committing.

#201 only affects 2.1.0 so I'm not sure why 1.6.0 is not working correctly.

**Comment by kanru** [comment]
Not sure if it's a Facebook issue. ibus-hangual also has similar behavior - preedit text disappear after commit.

**Comment by kanru** [comment]
Should be fixed in ibus-chewing 2.1.1, but I'm not sure what's wrong for 1.6.0

**Comment by ekangmonyet** [comment]
Well lot's of code has changed... I won't insist on getting F40 fixed.  2.1.1 is working for me too and I am happy about it, I don't mind closing this as completed.

EDIT: thanks a lot for your time too!
